### PR TITLE
Image sharing react setup

### PR DESCRIPTION
**Pull request by sagarayi**
Closes React #1 (Adding footer)

### CODE REVIEW

**Comment by sagarayi** [comment]
[React Issue 1: Add Footer](https://trello.com/c/mOVPfOsw/1-react-issue-1-add-footer)

File: app/javascript/image_sharing/components/Footer.js [hunk 1]
```diff
@@ -1,7 +1,26 @@
 import React, { Component } from 'react';
+import { Row, Col } from 'reactstrap';
+import PropTypes from 'prop-types';
 
 class Footer extends Component {
-  /* Implement your Footer component here */
+  static propTypes = {
```
Comment (zju2) [inline app/javascript/image_sharing/components/Footer.js:6]: We always want our commit as small as possible, it would be better if we split the Footer component out from this commit, and put it inside a separate commit.
Like "sa - Create Footer component", also, in React story, we put Component and Component test file into a same commit. So we may want to put `Footer.js` and `Footer.spec.js` into a new commit

File: app/javascript/image_sharing/test/FooterTest.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* eslint-env mocha */
+
+import { expect } from 'chai';
```
Comment (zju2) [inline app/javascript/image_sharing/test/FooterTest.js:3]: In React App, we name test file like this: `componentName.spec.js`
SO, we should rename this test to `Footer.spec.js`

Comment (zju2) [reply app/javascript/image_sharing/test/FooterTest.js:3]: And it would be better if we split this test file out, and put it with Footer component into a new commit

File: app/javascript/image_sharing/test/HeaderTest.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* eslint-env mocha */
+
```
Comment (zju2) [inline app/javascript/image_sharing/test/HeaderTest.js:2]: Same here, the test file should be renamed as `Header.spec.js`

File: app/javascript/image_sharing/components/App.js [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import React, { Component } from 'react';
+import { inject } from 'mobx-react';
```
Comment (zju2) [inline app/javascript/image_sharing/components/App.js:2]: Those test fixes should be included in previous commits, you may use `git commit --fixup` command to do commit fixing

File: app/javascript/image_sharing/components/App.js [hunk 5]
```diff
@@ -1,20 +1,18 @@
 import React, { Component } from 'react';
-import { inject } from 'mobx-react';
 import Header from './Header';
-import Footer from "./Footer";
+import Footer from './Footer';
```
Comment (zju2) [inline app/javascript/image_sharing/components/App.js:4]: our commits should be present chronologically so that reviewer can understand them easier,
 like here, we create Footer component first, then add this component to App.js, 
so we should put the commit creating Footer component before the commit adding Footer component to App.js.

File: app/javascript/image_sharing/components/Footer.js (hunk 1 above)
Comment (sagarayi) [reply app/javascript/image_sharing/components/Footer.js:6]: Sure will do it.

File: app/javascript/image_sharing/test/FooterTest.js (hunk 2 above)
Comment (sagarayi) [reply app/javascript/image_sharing/test/FooterTest.js:3]: ok sure

File: app/javascript/image_sharing/test/HeaderTest.js (hunk 3 above)
Comment (sagarayi) [reply app/javascript/image_sharing/test/HeaderTest.js:2]: ok

File: app/javascript/image_sharing/components/App.js (hunk 4 above)
Comment (sagarayi) [reply app/javascript/image_sharing/components/App.js:2]: Sure

File: app/javascript/image_sharing/components/App.js (hunk 5 above)
Comment (sagarayi) [reply app/javascript/image_sharing/components/App.js:4]: ok sure
